tests/Unit/Parallel/Test_TypeTraits.cpp

```diff
@@ -0,0 +1,43 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Parallel/TypeTraits.hpp"
+#include "tests/Unit/RunTests.hpp"
+
+static_assert(Parallel::is_array_proxy<CProxy_TestArrayChare>::value,
+              "Failed testing type trait is_array_proxy");
+static_assert(!Parallel::is_array_proxy<CProxy_TestChare>::value,
```

Comment:
Do we want to favor `not` over `!`? I personally use `not`, but don't have issues with `!`. This is related to us preferring `and` over `&&` and `or` over `||`. I'm looking for a poll here:

:+1:  = change to `not`
:-1:   = leave as `!`

@wthrowe @kidder @osheamonn @gsb76 @markscheel 

4/6 wins (we really need an odd number)